compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -228,6 +238,7 @@ impl DataFlowGraph {
         call_stack: CallStackId,
     ) -> InsertInstructionResult {
         if !self.is_handled_by_runtime(&instruction) {
+            warn!("Attempted to insert instruction not handled by runtime: {instruction:?}");
```

Comment:
By the way I inserted the same call [here](https://github.com/noir-lang/noir/blob/33cad9b4543825bb275a936bdfe320e9da804b54/compiler/noirc_evaluator/src/ssa/ir/dfg.rs#L217) when I saw that `insert_instruction_without_simplification` became public again to allow parsing SSA strings as-is, but in retrospect that's exactly why we might _not_ want to remove even these instructions, even if they would cause a panic later on: if the SSA has them, go ahead. So if you agree that this check should only be here, then we could remove the other instance.

Reply:
I've updated `insert_instruction_and_results_without_simplification` to panic if we hit this condition.